cicecore/cicedyn/infrastructure/io/io_pio2/ice_history_write.F90

```diff
@@ -192,73 +191,106 @@ subroutine ice_write_hist (ns)
       !-----------------------------------------------------------------
       ! define dimensions
       !-----------------------------------------------------------------
+      call pio_seterrorhandling(File, PIO_RETURN_ERROR)
 
-        if (hist_avg(ns) .and. .not. write_ic) then
-          status = pio_def_dim(File,'nbnd',2,boundid)
-        endif
+      if (hist_avg(ns) .and. .not. write_ic) then
+         call ice_pio_check(pio_def_dim(File,'nbnd',2,boundid), &
+              subname//' ERROR: defining dim nbnd with len 2',file=__FILE__,line=__LINE__)
+      endif
+
+      call ice_pio_check(pio_def_dim(File,'ni',nx_global,imtid), &
+           subname//' ERROR: defining dim ni',file=__FILE__,line=__LINE__)
```

Comment:
I guess this is ok. PIO has it's own error checking routines internally.

Reply:
Regarding your comment below, the pio error checking turns off the internal error handling for the most part.  There are some pio calls that don't return an error code, for those pio internal error is on.